hoomd/hpmc/tune/move_size.py

```diff
@@ -0,0 +1,333 @@
+
+        Can be set to False.
+        """
+        return self._tuned == 2
+
+    @tuned.setter
+    def tuned(self, value):
+        if not value:
+            self._tuned = 0
+        else:
+            raise ValueError("Cannot attempt to set tuned to True. "
+                             "Detach operation instead.")
+
+    def detach(self):
+        self._update_tunables_attr('integrator', None)
+        self._attached = False
+
+    def act(self, timestep=None):
+        """Tune move sizes.
+
+        Args:
+            timestep (:obj:`int`, optional): Current simulation timestep. Is
+                currently ignored.
+        """
+        if not self.tuned and self.is_attached:
+            tuned = self.solver.solve(self._tunables)
+            if tuned:
+                self._tuned += 1
+            else:
+                self._tuned = 0
+
+    def _update_tunables(self, *, new_moves=tuple(), new_types=tuple()):
+        tunables = self._tunables
+        tune_definitions = set(self._tunables)
+
+        # First filter out any move size tune definitions that don't match
+        # the new specification.
+        def filter_tunables(tunable):
+            return ((new_moves is None or tunable.attr in new_moves) and
+                    (new_types is None or tunable.type in new_types))
+
+        self._tunables = list(filter(filter_tunables, tunables))
+
+        # Add any move size tune definitions that are required by the new
+        # specification.
+        for move in new_moves:
+            for new_type in new_types:
+                move_definition = _MoveSizeTuneDefinition(
+                    move, new_type, self.target,
+                    (self._min_move_size, self.max_move_size))
+                if move_definition not in tune_definitions:
+                    self._tunables.append(move_definition)
+
+    def _update_tunables_attr(self, attr, value):
+        for tunable in self._tunables:
+            setattr(tunable, attr, value)
+
+
+class MoveSize(_InternalCustomTuner):
+    """Tunes HPMCIntegrator move sizes to targeted acceptance rate.
+
+    For most common creation of a `MoveSize` tuner see `MoveSize.secant_solver`
+    and `MoveSize.scale_solver` respectively.
+
+    Args:
+        trigger (hoomd.trigger.Trigger): ``Trigger`` to determine when to run
+            the tuner.
+        moves (list[str]): A list of types of moves to tune. Available options
+            are 'a' and 'd'.
+        target (float): The acceptance rate for trial moves that is desired. The
+            value should be between 0 and 1.
+        solver (`hoomd.tune.SolverStep`): A solver that tunes move sizes to
+            reach the specified target.
+        types (list[str]): A list of string particle types to tune the move
+            size for, defaults to None which upon attaching will tune all types
+            in the system currently.
+        max_move_size (float): The max value of move size to attempt.
```

Comment:
Is the maximum the same for both 'a' and 'd' when tuning both?

Reply:
Currently, yes. The `SolverStep` and `_TuneDefinition` are generic enough to allow for a difference, but I didn't how varied of maximum move sizes we wanted. Should we allow different maximums for each particle and move type?

Reply:
Yes, let's allow for this if it isn't too complex to provide the option. Depending on the diameter of the particles in the simulation users may need to provide very different maximum values. E.g. in the extreme case of a diameter 100 particle, the maximum move size for d might be 50 while (in 2D at least), the maximum move size for a is 2*pi.